internal/node/controller/controller2.go

```diff
@@ -64,38 +65,55 @@ func (c *controller2) startProcess(installationID string) error {
 		return errors.Wrap(err, "error proposing transition")
 	}
 
-	proc := transition.Process
-	err = c.processManager.StartProcess(c.ctx, proc, app.AppInstallation)
+	err = c.processManager.StartProcess(c.ctx, transition.Process.ID, app.AppInstallation)
 	if err != nil {
+		// Rollback the state change if the process start failed
+		_, err = c.db.ProposeTransitions([]hdb.Transition{
+			&node.ProcessStopTransition{
+				ProcessID: transition.Process.ID,
+			},
+		})
 		return errors.Wrap(err, "error starting process")
 	}
 	return nil
 }
 
 func (c *controller2) stopProcess(processID node.ProcessID) error {
-	err := c.processManager.StopProcess(c.ctx, processID)
-	if err != nil {
-		return errors.Wrap(err, "error stopping process")
+	procErr := c.processManager.StopProcess(c.ctx, processID)
+	// If there was no process found with this ID, continue with the state transition
+	// Otherwise this action failed, return an error without the transition
+	if procErr != nil && !errors.Is(procErr, process.ErrNoProcFound) {
+		// process.ErrNoProcFound is sometimes expected. In this case, still
+		// attempt to remove the process from the node state.
+		return procErr
 	}
 
-	// Only propose transitions if stopping the process succeeded
-	_, err = c.db.ProposeTransitions([]hdb.Transition{
-		&node.ProcessStopTransition{
-			ProcessID: processID,
-		},
-	})
+	state, err := c.getNodeState()
 	if err != nil {
 		return err
+	} else if _, ok := state.Processes[processID]; ok {
+		// Only propose transitions if the process exists in state
+		_, err = c.db.ProposeTransitions([]hdb.Transition{
```

Comment:
This should be redundant with the transition validation right?

Reply:
Removed